### DISCUSSION TITLE
What is the actual meanings of the abbrevations "ny" (direction), "nyP", and "nyPP" in the code?

### DISCUSSION THREAD

**Discussion by biergaizi** [thliebig/openEMS-Project#273] [Q&A Usage] [answered]
Although I've worked with the openEMS codebase for a year and I know basically what every part of it, but I never actually understood some of the naming convention used. In the code, the variable "ny" can be x, y, z, so it stands for "direction", but why is it represented by "ny"? Similarly, in the Mur's ABC code, another two related variables are "nyP" and "nyPP". These variables have something to do with polarization, but what does `nyP` and `nyPP` actually stand for?

Once I got an answer, I hope I can either rework with naming or at least document the convention used in the comments...

**Comment by biergaizi** [comment]
From the comments

    m_volt_nyP; //n+1 direction
    m_volt_nyPP; //n+2 direction

So I think `nyP` and `nyPP` stands for `ny_plus` and `ny_plusplus`, that makes sense, but what about the `y` in `ny`?

**Reply by thliebig** [reply]
I think you can just ignore that. It stems I think from the thesis and it was a n_μ [n_(μ+1), n_(μ+2)] which in the code I wrote as ny. Don't ask my why, its like from 20 years ago...

**Reply by thliebig** [reply, marked as answer]
in a sense "n" stood for "normal"-direction and μ, μ+1, μ+2 was the index/direction

**Reply by thliebig** [reply]
I guess it could be renamed to nu, nuP, nuPP ?  I tried to avoid just plain "n" (nP, nPP) as it is used often in loops etc.

**Reply by biergaizi** [reply]
After reading the code further, I think I know now. The `y` in `ny` probably means `Yee`, since this variable gives the direction with respect to the simulation box, on the other hand a plain `n` refers to the polarization direction of the vector field (not the box).

**Reply by thliebig** [reply]
I just noticed that I also used  this "ny" in the python interface, so we might just explain that it stands for direction in space (usually 0,1 or 2 for x,y or z)
